src/org/sosy_lab/java_smt/SolverContextFactory.java

```diff
@@ -55,6 +56,7 @@ public enum Solvers {
     MATHSAT5,
     SMTINTERPOL,
     Z3,
+    Z3LEGACY,
```

Comment:
@kfriedberger @PhilippWendler @leventeBajczi  i kinda feel like Z3 legacy is a little unspecific. Since the version is fixed, should we use the actual version? I.e. `Z3-3.5`?

Reply:
The solver is able to provide its version string anyway. As it is a version of Z3 with interpolation support, we might name it `Z3_INTERPOLATION` or simply `Z3_3_5` . Special chars like dots or hyphens are not allowed in Java enums.

The JavaSMT enum it independent of the native library, so the library can be called `z3Legacy` or similar.